File: python/fizzbuzz.py
def advanced(divisor_mappings):
    output = []
    found = False
    line = ''
    keys = divisor_mappings.keys()
    divisors = []
    for key in keys:
        divisors.append(int(key))
    for i in range(1, 101):
        found = False
        line = ''
        for j in range(0, len(divisors)):
            if i % divisors[j] == 0:
                found = True
                line += divisor_mappings[keys[j]]
        if not found:
            line += str(i)
        output.append(str(i) + ': ' + line)
    return "\n".join(output)


if __name__ == '__main__':
    print advanced({
        '2': 'second',
        '3': 'Fizz',
        '5': 'Buzz',
        '6': 'sixth'
    })

File: c++/fizzbuzz.cpp
#include <sstream>
#include <iostream>
#include <map>

using namespace std;


map<int,string> fizzBuzz(map<int,string> mymap){
    map<int,string> output;
    bool found = false;
    string newLongLine;

    for(int i = 1; i <101; i++) {
        found = false;
        newLongLine="";
        for(map<int,std::string>::iterator it=mymap.begin(); it!=mymap.end(); ++it) {
            if(i %  it->first == 0){
                found = true;
                newLongLine += it->second;
            }
        }
        if(!found) {
          std::ostringstream ss;
          ss << i;
          newLongLine = ss.str();
        }

        output.insert(pair<int,string>(i,newLongLine));
    }

    return output;
}

int main ()
{
    //Creates Map
    map<int,string> mymap;
    mymap[2] = "second";
    mymap[3] = "Fizz";
    mymap[5] = "Buzz";
    mymap[6] = "sixth";

    mymap = fizzBuzz(mymap);


    //Prints Map
    for (map<int,std::string>::iterator it=mymap.begin(); it!=mymap.end(); ++it)
        std::cout<<it->first<<':'<<it->second<<std::endl;


    return 0;
}
